maas/client/viscera/machines.py

```diff
@@ -171,6 +172,10 @@ class MachineType(ObjectType):
         data = await cls._handler.read(system_id=system_id)
         return cls(data)
 
+    async def details(cls, system_id):
+        data = await cls._handler.details(system_id=system_id)
+        return bson.loads(data)
```

Comment:
Since this takes exactly one system_id this would be better to place on the Machine object instead of the Machines object.

This would allow you to:

```
machine = client.machines.list()[0]
machine.details()
```

Simply moving this to the `Machine` handler using:

```
async def details(self):
    data = await type(self)._origin.Machines.details(system_id=self.system_id)
    return bson.loads(data)
```

You also need to add a unit tests for this new method.